docs/design/node-pool/package-based-lcm-node-pools.md

```diff
@@ -0,0 +1,59 @@
+
+```bash
+tanzu cluster node-pool delete tkg-wc-vsphere -n md–0
+```
+
+## List
+
+List retrieves the MachineDeployment resources as in the non clusterclass based case and matches these resources against the machine deployments in the cluster topology. This allows the list API to return the same MachineDeployment resources with the names updated to match the machine deployments defined in the cluster topology.
+
+```bash
+tanzu cluster node-pool list tkg-wc-vsphere
+  NAME  NAMESPACE  PHASE      REPLICAS  READY  UPDATED  UNAVAILABLE
+  md-0  default    Ready      1         1      1        0
+```
+
+## Properties on Node Pools in ClusterClass based Clusters
+
+Properties that can be updated on node pools in clusterclass based clusters can be broken down into two types. Direct properties of the machine deployment topology resources and variables overrides. Direct properties, like replica count, modify the machine deployment topolgy direct. Variable overrides rely on the definition of variables at the clusterclass level. When deploying a cluster based on a clusterclass, the cluster definition must specify a number of these variables, which apply to all machine deployments by default. The machine deployment topology allows these variables to be overridden on a machine deployment basis.
+
+## Set (Update)
+
+When the Set API is called using the name of an existing node pool the update will only modify the replica count and labels of the named node pool, if provided. Any other provided options will be ignored. Specifically the cluster's topology will be update with a new replica count and a nodePoolLabels variable override if these values are provided.
+
+## Set (Create)
+
+Creating node pools can follow one of two paths. A user can pass a base machine deployment that is used as the starting point for the new node pool. A deep copy of the named base machine deployment acts as a basis on which the user passed properties will be layered. In general this means all of the variable overrides of the copied machine deployment will be updated with new values provided by the user. This is different from the case where a base machine deployment is not provided. In this case, variable overrides will be copied from the global variable definitions defined in the cluster topology and then the user provided values will be layered on those. Additionally when a user does not provide a base machine deployment, they must provide a workerClass and tkrResolver definition. TKG ClusterClasses have a `tkg-worker` class defined by default, and should be used unless a user have added their own worker class definition. The tkrResolver is the value of the tkr-resolver annotation.
+
+Using a base machine deployment to create a new node pool.
+
+```bash
+tanzu cluster node-pool set tkg-wc-vsphere -f /path/to/node/pool/def –-base-machine-deployment md-0
+```
+
+```yaml
```

Comment:
thanks for a examples. nit: this and the yaml in the next section looks like command output, but I think you meant them to be contents of input file `/path/to/node/pool/def `
maybe a header above the yaml will make it clear.

Reply:
done